pkg/app/ops/insightcollector/collector.go

```diff
@@ -13,3 +13,354 @@
 // limitations under the License.
 
 package insightcollector
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/pipe-cd/pipe/pkg/datastore"
+	"github.com/pipe-cd/pipe/pkg/filestore"
+	"github.com/pipe-cd/pipe/pkg/insightstore"
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+var aggregateKinds = []model.InsightMetricsKind{
+	model.InsightMetricsKind_DEPLOYMENT_FREQUENCY,
+	model.InsightMetricsKind_CHANGE_FAILURE_RATE,
+}
+
+// InsightCollector implements the behaviors for the gRPC definitions of InsightCollector.
+type InsightCollector struct {
+	applicationStore datastore.ApplicationStore
+	deploymentStore  datastore.DeploymentStore
+	insightstore     insightstore.Store
+	logger           *zap.Logger
+}
+
+// NewInsightCollector creates a new InsightCollector instance.
+func NewInsightCollector(
+	ds datastore.DataStore,
+	fs filestore.Store,
+	logger *zap.Logger,
+) *InsightCollector {
+	a := &InsightCollector{
+		applicationStore: datastore.NewApplicationStore(ds),
+		deploymentStore:  datastore.NewDeploymentStore(ds),
+		insightstore:     insightstore.NewStore(fs),
+		logger:           logger.Named("insight-collector"),
+	}
+	return a
+}
+
+func (i *InsightCollector) Run(ctx context.Context) error {
+	now := time.Now().UTC()
+	maxUpdateAt := now.Unix()
+
+	for {
+		listAppsPageSize := 50
+
+		// TODO: When [#1195](https://github.com/pipe-cd/pipe/issues/1195) is resolved, modify to use page option instead of filter with updatedAt.
```

Comment:
And sorry for the not enough description, but this link should be on the second line (The first line will be used for the title of PR to be opened).

Reply:
I've fixed it.
Isn't that different from the comment you are imagining?